Vulkan-Hpp/tests/ArrayWrapper/ArrayWrapper.cpp:36:32: error: call to constructor of 'vk::ArrayWrapper1D<char, 20>' is ambiguous    36 |   vk::ArrayWrapper1D<char, 20> aw2( "foobah" );

```
~/Vulkan-Hpp $ git pull && git submodule foreach git pull && date
remote: Enumerating objects: 16, done.
remote: Counting objects: 100% (16/16), done.
remote: Total 16 (delta 15), reused 16 (delta 15), pack-reused 0
Unpacking objects: 100% (16/16), 13.42 KiB | 44.00 KiB/s, done.
From https://github.com/KhronosGroup/Vulkan-Hpp
   3e0d722..9c15a1e  main       -> origin/main
* [new tag]         v1.3.269   -> v1.3.269
Fetching submodule Vulkan-Headers
From https://github.com/KhronosGroup/Vulkan-Headers
   f4bfcd8..374f9fd  main       -> origin/main
* [new branch]      449-genvk  -> origin/449-genvk
* [new tag]         v1.3.269   -> v1.3.269
Updating 3e0d722..9c15a1e
Fast-forward
Vulkan-Headers                         |   2 +-
vulkan/vulkan.cppm                     |  42 +++
vulkan/vulkan.hpp                      | 157 ++++++++++-
vulkan/vulkan_enums.hpp                |  42 ++-
vulkan/vulkan_extension_inspection.hpp |  36 ++-
vulkan/vulkan_funcs.hpp                | 298 ++++++++++++++++++++
vulkan/vulkan_handles.hpp              | 325 +++++++++++++++++++++
vulkan/vulkan_hash.hpp                 | 158 +++++++++++
vulkan/vulkan_raii.hpp                 | 356 +++++++++++++++++++++++
vulkan/vulkan_shared.hpp               |  23 ++
vulkan/vulkan_static_assertions.hpp    |  67 +++++
vulkan/vulkan_structs.hpp              | 982 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
vulkan/vulkan_to_string.hpp            |  43 +++
vulkan/vulkansc.hpp                    |   2 +-
14 files changed, 2517 insertions(+), 16 deletions(-)
Entering 'Vulkan-Headers'
Updating f4bfcd8..374f9fd
Fast-forward
README.md                                      |    6 +-
include/vulkan/vulkan.cppm                     |   48 +
include/vulkan/vulkan.hpp                      |  181 +++-
include/vulkan/vulkan_core.h                   |  136 ++-
include/vulkan/vulkan_enums.hpp                |   42 +-
include/vulkan/vulkan_extension_inspection.hpp |   34 +-
include/vulkan/vulkan_funcs.hpp                |  298 ++++++
include/vulkan/vulkan_handles.hpp              |  336 +++++++
include/vulkan/vulkan_hash.hpp                 |  158 +++
include/vulkan/vulkan_raii.hpp                 |  350 +++++++
include/vulkan/vulkan_shared.hpp               |   70 +-
include/vulkan/vulkan_static_assertions.hpp    |   67 ++
include/vulkan/vulkan_structs.hpp              |  982 +++++++++++++++++++
include/vulkan/vulkan_to_string.hpp            |   43 +
registry/validusage.json                       | 2850 ++++++++++++++++++++++++++++++++++---------------------
registry/vk.xml                                |  153 ++-
16 files changed, 4619 insertions(+), 1135 deletions(-)
Entering 'glfw'
Already up to date.
Entering 'glm'
Already up to date.
Entering 'glslang'
remote: Enumerating objects: 17, done.
remote: Counting objects: 100% (17/17), done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 17 (delta 6), reused 5 (delta 3), pack-reused 0
Unpacking objects: 100% (17/17), 9.29 KiB | 559.00 KiB/s, done.
From https://github.com/KhronosGroup/glslang
   ebb0a8b2..7fa0731a  main       -> origin/main
* [new branch]        dependabot/github_actions/actions/checkout-4.1.1 -> origin/dependabot/github_actions/actions/checkout-4.1.1
* [new branch]        dependabot/github_actions/github/codeql-action-2.22.4 -> origin/dependabot/github_actions/github/codeql-action-2.22.4
Updating ebb0a8b2..7fa0731a
Fast-forward
SPIRV/CInterface/spirv_c_interface.cpp | 9 +--------
1 file changed, 1 insertion(+), 8 deletions(-)
Entering 'tinyxml2'
Already up to date.
Mon Oct 23 08:47:37 PDT 2023
~/Vulkan-Hpp $ cat RunCmakeSamples.sh
sed 's/STREQUAL "Linux"/MATCHES "Linux|Android"/g' glfw/src/CMakeLists.txt -i''
sed 's/STREQUAL "Linux"/MATCHES "Linux|Android"/g' glfw/src/x11_init.c -i''
echo cmake \
  --fresh \
  -B build \
  -DVULKAN_HPP_SAMPLES_BUILD=ON \
  -DVULKAN_HPP_TESTS_BUILD=ON \
  -DVULKAN_HPP_BUILD_WITH_LOCAL_VULKAN_HPP=ON \
  -DENABLE_OPT=1 \
  -DALLOW_EXTERNAL_SPIRV_TOOLS=ON \
  -G Ninja \
  &&
#sed 's/-Werror/-Wno-error/g' CMakeLists.txt -i'' && #/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxy/ArrayProxy.cpp:39:25: error: unused variable 'ap0' [-Werror,-Wunused-variable]
sed 's/-W\(no\-\)\?error )/-W\1error -Wno-c++20-extensions -std=c++20)/g' CMakeLists.txt -i'' && #silence warnings about <variant>/c++17, inline variables/c++20 designators
cmake --build build --parallel2 --clean-first
~/Vulkan-Hpp $ ./RunCmakeSamples.sh
cmake -B build -DVULKAN_HPP_SAMPLES_BUILD=ON -DVULKAN_HPP_TESTS_BUILD=ON -DVULKAN_HPP_BUILD_WITH_LOCAL_VULKAN_HPP=ON -DENABLE_OPT=1 -DALLOW_EXTERNAL_SPIRV_TOOLS=ON -G Ninja
[0/1] Re-running CMake...
--  Found clang-format version <17.0.3>.
--  Using .clang-format version 15.
-- GLM: Version 0.9.9.9
-- Including X11 support
-- Google Mock was not found - tests based on that will not build
-- Trying to find local SPIR-V tools
-- optimizer enabled
-- Configuring done (6.7s)
-- Generating done (0.1s)
-- Build files have been written to: /data/data/com.termux/files/home/Vulkan-Hpp/build
[220/257] Building CXX object tests/ArrayProxyNoTempor...ayProxyNoTemporaries.dir/ArrayProxyNoTemporaries.cpp.
FAILED: tests/ArrayProxyNoTemporaries/CMakeFiles/ArrayProxyNoTemporaries.dir/ArrayProxyNoTemporaries.cpp.o
/data/data/com.termux/files/usr/bin/c++ -DCLANG_FORMAT_EXECUTABLE=\"/data/data/com.termux/files/usr/bin/clang-format\" -DVK_USE_PLATFORM_XCB_KHR -DVULKAN_HPP_DISPATCH_LOADER_DYNAMIC=1 -I/data/data/com.termux/files/home/Vulkan-Hpp -I/data/data/com.termux/files/home/Vulkan-Hpp/Vulkan-Headers/include -I/data/data/com.termux/files/home/Vulkan-Hpp/glm -I/data/data/com.termux/files/home/Vulkan-Hpp/glfw/include -I/data/data/com.termux/files/home/Vulkan-Hpp/glslang/glslang/.. -I/data/data/com.termux/files/home/Vulkan-Hpp/build/include -I/data/data/com.termux/files/home/Vulkan-Hpp/glslang/SPIRV/.. -I/data/data/com.termux/files/home/Vulkan-Hpp/glslang/SPIRV/../External -O2 -g -DNDEBUG -std=gnu++11 -Wall -Wextra -pedantic -Werror -Wno-c++20-extensions -std=c++20 -MD -MT tests/ArrayProxyNoTemporaries/CMakeFiles/ArrayProxyNoTemporaries.dir/ArrayProxyNoTemporaries.cpp.o -MF tests/ArrayProxyNoTemporaries/CMakeFiles/ArrayProxyNoTemporaries.dir/ArrayProxyNoTemporaries.cpp.o.d -o tests/ArrayProxyNoTemporaries/CMakeFiles/ArrayProxyNoTemporaries.dir/ArrayProxyNoTemporaries.cpp.o -c /data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:90:38: error: unused variable 'ap0' [-Werror,-Wunused-variable]
   90 |     vk::ArrayProxyNoTemporaries<int> ap0 = nullptr;
      |                                      ^~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:111:44: error: unused variable 'ap2' [-Werror,-Wunused-variable]
  111 |     vk::ArrayProxyNoTemporaries<const int> ap2 = i0;
      |                                            ^~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:113:44: error: unused variable 'ap3' [-Werror,-Wunused-variable]
  113 |     vk::ArrayProxyNoTemporaries<const int> ap3 = i1;
      |                                            ^~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:141:44: error: unused variable 'ap6' [-Werror,-Wunused-variable]
  141 |     vk::ArrayProxyNoTemporaries<const int> ap6 = ia0;
      |                                            ^~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:143:44: error: unused variable 'ap7' [-Werror,-Wunused-variable]
  143 |     vk::ArrayProxyNoTemporaries<const int> ap7 = ia1;
      |                                            ^~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:179:38: error: unused variable 'ap8' [-Werror,-Wunused-variable]
  179 |     vk::ArrayProxyNoTemporaries<int> ap8 = sa0;
      |                                      ^~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:187:44: error: unused variable 'ap12' [-Werror,-Wunused-variable]
  187 |     vk::ArrayProxyNoTemporaries<const int> ap12 = sa0;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:189:44: error: unused variable 'ap13' [-Werror,-Wunused-variable]
  189 |     vk::ArrayProxyNoTemporaries<const int> ap13 = sa1;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:191:44: error: unused variable 'ap14' [-Werror,-Wunused-variable]
  191 |     vk::ArrayProxyNoTemporaries<const int> ap14 = sa2;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:193:44: error: unused variable 'ap15' [-Werror,-Wunused-variable]
  193 |     vk::ArrayProxyNoTemporaries<const int> ap15 = sa3;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:213:38: error: unused variable 'ap16' [-Werror,-Wunused-variable]
  213 |     vk::ArrayProxyNoTemporaries<int> ap16 = sv0;
      |                                      ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:219:44: error: unused variable 'ap18' [-Werror,-Wunused-variable]
  219 |     vk::ArrayProxyNoTemporaries<const int> ap18 = sv0;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:221:44: error: unused variable 'ap19' [-Werror,-Wunused-variable]
  221 |     vk::ArrayProxyNoTemporaries<const int> ap19 = sv1;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:269:44: error: unused variable 'ap24' [-Werror,-Wunused-variable]
  269 |     vk::ArrayProxyNoTemporaries<const int> ap24 = {};
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:274:44: error: unused variable 'ap26' [-Werror,-Wunused-variable]
  274 |     vk::ArrayProxyNoTemporaries<const int> ap26 = il1;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:276:44: error: unused variable 'ap27' [-Werror,-Wunused-variable]
  276 |     vk::ArrayProxyNoTemporaries<const int> ap27 = il2;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:278:44: error: unused variable 'ap28' [-Werror,-Wunused-variable]
  278 |     vk::ArrayProxyNoTemporaries<const int> ap28 = il3;
      |                                            ^~~~
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp:280:44: error: unused variable 'ap29' [-Werror,-Wunused-variable]
  280 |     vk::ArrayProxyNoTemporaries<const int> ap29 = il4;
      |                                            ^~~~
18 errors generated.
[221/257] Building CXX object tests/ArrayWrapper/CMakeFiles/ArrayWrapper.dir/ArrayWrapper.cpp.o
FAILED: tests/ArrayWrapper/CMakeFiles/ArrayWrapper.dir/ArrayWrapper.cpp.o
/data/data/com.termux/files/usr/bin/c++ -DCLANG_FORMAT_EXECUTABLE=\"/data/data/com.termux/files/usr/bin/clang-format\" -DVK_USE_PLATFORM_XCB_KHR -DVULKAN_HPP_DISPATCH_LOADER_DYNAMIC=1 -I/data/data/com.termux/files/home/Vulkan-Hpp -I/data/data/com.termux/files/home/Vulkan-Hpp/Vulkan-Headers/include -I/data/data/com.termux/files/home/Vulkan-Hpp/glm -I/data/data/com.termux/files/home/Vulkan-Hpp/glfw/include -I/data/data/com.termux/files/home/Vulkan-Hpp/glslang/glslang/.. -I/data/data/com.termux/files/home/Vulkan-Hpp/build/include -I/data/data/com.termux/files/home/Vulkan-Hpp/glslang/SPIRV/.. -I/data/data/com.termux/files/home/Vulkan-Hpp/glslang/SPIRV/../External -O2 -g -DNDEBUG -std=gnu++11 -Wall -Wextra -pedantic -Werror -Wno-c++20-extensions -std=c++20 -MD -MT tests/ArrayWrapper/CMakeFiles/ArrayWrapper.dir/ArrayWrapper.cpp.o -MF tests/ArrayWrapper/CMakeFiles/ArrayWrapper.dir/ArrayWrapper.cpp.o.d -o tests/ArrayWrapper/CMakeFiles/ArrayWrapper.dir/ArrayWrapper.cpp.o -c /data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayWrapper/ArrayWrapper.cpp
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayWrapper/ArrayWrapper.cpp:36:32: error: call to constructor of 'vk::ArrayWrapper1D<char, 20>' is ambiguous
   36 |   vk::ArrayWrapper1D<char, 20> aw2( "foobah" );
      |                                ^    ~~~~~~~~
/data/data/com.termux/files/home/Vulkan-Hpp/vulkan/vulkan.hpp:98:29: note: candidate constructor [with B = char, $1 = 0]
   98 |     VULKAN_HPP_CONSTEXPR_14 ArrayWrapper1D( std::string const & data ) VULKAN_HPP_NOEXCEPT
      |                             ^
/data/data/com.termux/files/home/Vulkan-Hpp/vulkan/vulkan.hpp:105:29: note: candidate constructor [with B = char, $1 = 0]
  105 |     VULKAN_HPP_CONSTEXPR_14 ArrayWrapper1D( std::string_view data ) VULKAN_HPP_NOEXCEPT
      |                             ^
/data/data/com.termux/files/home/Vulkan-Hpp/tests/ArrayWrapper/ArrayWrapper.cpp:44:31: error: call to constructor of 'vk::ArrayWrapper1D<char, 5>' is ambiguous
   44 |   vk::ArrayWrapper1D<char, 5> aw4( "foobah" );
      |                               ^    ~~~~~~~~
/data/data/com.termux/files/home/Vulkan-Hpp/vulkan/vulkan.hpp:98:29: note: candidate constructor [with B = char, $1 = 0]
   98 |     VULKAN_HPP_CONSTEXPR_14 ArrayWrapper1D( std::string const & data ) VULKAN_HPP_NOEXCEPT
      |                             ^
/data/data/com.termux/files/home/Vulkan-Hpp/vulkan/vulkan.hpp:105:29: note: candidate constructor [with B = char, $1 = 0]
  105 |     VULKAN_HPP_CONSTEXPR_14 ArrayWrapper1D( std::string_view data ) VULKAN_HPP_NOEXCEPT
      |                             ^
2 errors generated.
ninja: build stopped: subcommand failed.
```

--- a/tests/ArrayProxy/ArrayProxy.cpp
+++ b/tests/ArrayProxy/ArrayProxy.cpp
@@ -22,7 +22,7 @@
 #elif defined( __GNUC__ )
 #  pragma GCC diagnostic ignored "-Wunused-but-set-variable"
 #else
-// unknow compiler... just ignore the warnings for yourselves ;)
+// unknown compiler... just ignore the warnings for yourselves ;)
 #endif
 
 #include <iostream>

--- a/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp
+++ b/tests/ArrayProxyNoTemporaries/ArrayProxyNoTemporaries.cpp
@@ -17,12 +17,12 @@
 
 #if defined( _MSC_VER )
 // no need to ignore any warnings with MSVC
+#elif defined( __clang__ )
+#  pragma clang diagnostic ignored "-Wunused-variable"
 #elif defined( __GNUC__ )
-#  if ( 9 <= __GNUC__ )
-#    pragma GCC diagnostic ignored "-Winit-list-lifetime"
-#  endif
+#  pragma GCC diagnostic ignored "-Wunused-but-set-variable"
 #else
-// unknow compiler... just ignore the warnings for yourselves ;)
+// unknown compiler... just ignore the warnings for yourselves ;)
 #endif
 
 #include <iostream>

--- a/tests/ArrayWrapper/ArrayWrapper.cpp
+++ b/tests/ArrayWrapper/ArrayWrapper.cpp
@@ -33,7 +33,9 @@ int main( int /*argc*/, char ** /*argv*/ )
 
   // s1 = aw1;  // 'operator =' is ambiguous
 
-  vk::ArrayWrapper1D<char, 20> aw2( "foobah" );
+  std::string foobah = "foobah";
+
+  vk::ArrayWrapper1D<char, 20> aw2( foobah );
   f( aw2 );
 
   vk::ArrayWrapper1D<char, 5> aw3( { 'f', 'o', 'o', 'b', 'a', 'h' } );
@@ -41,7 +43,7 @@ int main( int /*argc*/, char ** /*argv*/ )
   assert( s3.length() == 5 );
   std::cout << "<" << s3 << ">" << std::endl;
 
-  vk::ArrayWrapper1D<char, 5> aw4( "foobah" );
+  vk::ArrayWrapper1D<char, 5> aw4( foobah );
   std::string                 s4 = aw4;
   assert( s4.length() == 5 );
 

--- a/tests/UniqueHandle/UniqueHandle.cpp
+++ b/tests/UniqueHandle/UniqueHandle.cpp
@@ -27,6 +27,19 @@ static std::string EngineName = "Vulkan.hpp";
 template <typename T>
 class MyAllocator : public std::allocator<T>
 {
+public:
+  MyAllocator() = default;
+
+  template <class U>
+  MyAllocator( const MyAllocator<U> & /*unused*/ )
+  {
+  }
+
+  template <class U>
+  struct rebind
+  {
+    using other = MyAllocator<U>;
+  };
 };
 
 vk::UniqueDescriptorSetLayout createDescriptorSetLayoutUnique( vk::Device const &                                                                  device,

